www/src/AdminBundle/Model/ImportTwine.php

```diff
@@ -0,0 +1,592 @@
+<?php
+/* Copyright (C) 2016 Michael Giesler
+ *
+ * This file is part of Dembelo.
+ *
+ * Dembelo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Dembelo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License 3 for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License 3
+ * along with Dembelo. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/**
+ * @package AdminBundle
+ */
+
+namespace AdminBundle\Model;
+
+use DembeloMain\Document\Importfile;
+use DembeloMain\Document\Textnode;
+use DembeloMain\Model\Repository\TextNodeRepositoryInterface;
+use DembeloMain\Model\Repository\TopicRepositoryInterface;
+use Doctrine\ODM\MongoDB\DocumentRepository;
+use Exception;
+use phpDocumentor\Reflection\DocBlock\Tags\Method;
+
+/**
+ * Class ImportTwine
+ * @package AdminBundle\Model
+ */
+class ImportTwine
+{
+    /**
+     * @var TextnodeRepositoryInterface
+     */
+    private $textnodeRepository;
+
+    /**
+     * @var TopicRepositoryInterface
+     */
+    private $topicRepository;
+
+    private $xmlParser;
+
+    /**
+     * @var Importfile
+     */
+    private $importfile;
+
+    /**
+     * @var \DembeloMain\Document\Textnode
+     */
+    private $textnode = null;
+    private $topicId = null;
+    private $nodeNameMapping = [];
+    private $accessSet = false;
+    private $twineId;
+
+    private $twineRelevant = false;
+    private $twineStartnodeId = -1;
+    private $twineTextnodeName = null;
+    private $twineText = false;
+
+    private $textnodeMapping;
+
+    /**
+     * ImportTwine constructor.
+     * @param TextnodeRepositoryInterface $textnodeRepository
+     * @param TopicRepositoryInterface    $topicRepository
+     */
+    public function __construct(TextNodeRepositoryInterface $textnodeRepository, TopicRepositoryInterface $topicRepository)
+    {
+        $this->textnodeRepository = $textnodeRepository;
+        $this->topicRepository = $topicRepository;
+    }
+
+    /**
+     * main method, starts the import process
+     *
+     * @param Importfile $importfile
+     * @return bool
+     * @throws Exception
+     */
+    public function run(Importfile $importfile)
+    {
+        $this->importfile = $importfile;
+
+        if (is_null($this->importfile->getLicenseeId())) {
+            throw new Exception('no licensee available');
```

Comment:
Ja, du hast Recht. Dieses Refactoring übersteigt aber das, was ich in diesem Issue erreichen will.
Refactoring != komplett neu machen